app/views/layouts/_navigation.html.erb

```diff
@@ -0,0 +1,26 @@
+    <nav class="nav">
+      <%= link_to 'Stay in touch', root_path %>
+      <div class="menu">
+        <div class="menu-item">
+          <%= menu_link_to 'Timeline', posts_path %>
+        </div>
+        <div class="menu-item">
+          <%= menu_link_to 'All users', users_path %>
+        </div>
+      </div>
+      <% if current_user %>
+        <%= link_to 'Sign out', destroy_user_session_path, method: :delete %>
+      <% else %>
+        <%= link_to 'Sign in', user_session_path %>
+      <% end %>
+    </nav>
+    <% if notice.present? %>
+      <div class="notice">
+        <p> <%= notice %> </p>
+      </div>
+    <% end %>
+    <% if alert.present? %>
+      <div class="alert">
+        <p> <%= alert %> </p>
+      </div>
+    <% end %>
```

Comment:
You didn't address the previous reviewer's comment about the logic in the views. Please create a helpers folder where you take care of this and use it in here.
There are other cases that you also need to take care of. Please go over your views and remove the logic from all of them.